docusaurus/docs/how-to-guides/data-source-plugins/add-migration-handler-for-backend.md

```diff
@@ -0,0 +1,216 @@
+
+:::
+
+### Step 1 (optional): Add a query schema
+
+First of all, plugins don't need to have strongly typed queries. While this lowers the barrier for plugin development, plugins that don't define types are harder to scale and maintain. The first step in this guide is to add the required files to define the plugin query.
+
+See the following example: [grafana-plugin-examples#400](https://github.com/grafana/grafana-plugin-examples/pull/400). As you can see, there are multiple files to create. These files will be used for both generating OpenAPI documentation and validating that the received queries are valid but it's a feature still in progress that isn't available yet.
+
+Create these files:
+
+- `query.go`: This file defines the Golang types for your query. For automatic migrations to work, it's important that your query extends the new `v0alpha1.CommonQueryProperties`. After that, just define your query custom properties.
+- `query_test.go`: This test file is both used to check that all the JSON files are up to date with the query model and to generate them. The first time you execute the test, it will generate these files (so take into account that `query.types.json` needs to exist, even if it's empty).
+- `query.*.json`: Automatically generated files. These schemas can be used for OpenAPI documentation.
+
+### Step 2: Changing the query model
+
+:::note
+
+For a complete example of how to add a query migration (steps 2, 3 and 4), refer to the code for [experimental APIs](https://github.com/grafana/grafana-plugin-examples/pull/403/files) or [stable APIs](https://github.com/grafana/grafana-plugin-examples/pull/407).
+
+:::
+
+Once your plugin has its own schemas, you can start introducing model changes. Since queries within the major version (or same API version) need to be compatible, you have to maintain a reference to the legacy data format. This reference also helps you to enable an easy migration path.
+
+For example, let's assume that you want to change the query format of your plugin and the `Multiplier` property that you were using is changing to `Multiply` like so:
+
+```patch
+ type DataQuery struct {
+        v0alpha1.CommonQueryProperties
+
+-       // Multiplier is the number to multiply the input by
++       // Multiply is the number to multiply the input by
++       Multiply int `json:"multiply,omitempty"`
++
++       // Deprecated: Use Multiply instead
+        Multiplier int `json:"multiplier,omitempty"`
+ }
+```
+
+In this example, you can regenerate the schema running the test in `query_test.go` so your new data type will become ready to be used.
+
+Note that we haven't yet made a breaking change because all new parameters (in this case `Multiply`) are marked as optional. Also, we haven't modified any of the business logic, so everything should work as before, using the deprecated property. We're going to actually make a breaking change in the next step.
```

Comment:
```suggestion
Note that there is not yet a breaking change because all the new parameters (in this case `Multiply`) are marked as optional. Also, none of the other business logic has been modified, so everything should work as before, using the deprecated property. In the next step, there is actually a breaking change. 
```